kratos/python_scripts/restart_utility.py

```diff
@@ -192,21 +193,16 @@ def CreateOutputFolder(self):
 
     def GetRestartFiles(self):
         """Return a dictionary of stepID - restart_file_list dictionary that stores sets of restart files for each step."""
-        restart_files = {}
-        if os.path.isdir(self.__GetFolderPathSave()):
-            number_of_restart_files = 0
-
-            with os.scandir(path=self.__GetFolderPathSave()) as contents:
-                for entry in contents:
-                    if self.__IsRestartFile(entry):
-
-                        label = self.__ExtractFileLabel(entry.name) # Get thread ID and step ID
-                        if label[1] in restart_files:               # Check if this step has entries already
-                            restart_files[label[1]].append( entry.name )
-                        else:
-                            restart_files[label[1]] = [ entry.name ]
-
-                        number_of_restart_files += 1
+        restart_path    = Path( self.__GetFolderPathSave() )
+        restart_files   = {}
+        if restart_path.is_dir():
+            for entry in restart_path.glob("**/*"):
```

Comment:
If that is a forward slash hardcoded, might be problematic on various OSs. See `os.path.sep` (or equivalent ) instead.

Reply:
[pathlib](https://github.com/python/cpython/blob/master/Lib/pathlib.py) instantiates specialized classes for the OS the interpreter is running in, and replaces separators with the correct ones internally whenever a function with path arguments is called. Nevertheless I can change it to use the generalized separators if you think it makes it more readable.

Reply:
this is fine, thats why I suggested to use pathlib ;)